src/viur/core/modules/user.py

```diff
@@ -1217,11 +1217,9 @@ def secondFactorProviderByClass(self, cls) -> UserSecondFactorAuthentication:
         return getattr(self, "f2_%s" % cls.__name__.lower())
 
     def getCurrentUser(self):
-        # May be a deferred task
-        if not (session := current.session.get()):
-            return None
+        session = current.session.get()
 
-        if user := session.get("user"):
+        if session and (user := session.get("user")):
```

Comment:
Am I blind? But the way I see it, there's no logical change here, just a "design change", right?

Reply:
right!